### ISSUE TITLE
Tempo(gram) display with musical notation

### ISSUE THREAD

**Issue by bmcfee** [librosa/librosa#1426]
**Is your feature request related to a problem? Please describe.**
Our tempo axes represent data on a BPM axis, which is great if you want to understand the (local) speed of a recording.
This is analogous to our frequency axis plots (eg, `cqt_hz`).
We also provide pitch-based axes (`cqt_note`, `cqt_svara_*`) which use the same coordinate system but notate the axes musically instead of physically.
This got me thinking: why not do the same thing for tempo?

**Describe the solution you'd like**
We could provide a couple of new axis modes, eg `tempo_notes`, which would additionally require a `tempo=` parameter to locate the position of the quarter note.
From there, we could add tick marks at the musically related positions (eighth note, half note, etc; and if desired, dotted notes and triplets).
Ticks could be decorated with note heads instead of tempo in BPM.  This would provide a more musically meaningful visualization.

Here's an old prototype that I coded up years ago when reimplementing some work of Peeters (2005) and Prockup (2015):

![image](https://user-images.githubusercontent.com/1190540/146947104-69e35ac7-bc44-4ea3-9e0c-3783ffe4fad6.png)

**Describe alternatives you've considered**
I've done this kind of plot by hand before, and it's not exactly difficult.  But we may as well make it easy, right?

**Additional context**
The addition of a `tempo=` parameter to `specshow` has precedent in the setting of `Sa=` for svara notation.

I expect this kind of plot would be more useful for a tempo histogram than for a time-varying tempogram, but the same mechanism could be used in both cases:
![image](https://user-images.githubusercontent.com/1190540/146946827-164816d3-e09f-4ea3-8861-b51eb9240e7c.png)

The above examples were done using latex rendering, but I think it could be replaced by unicode symbols (♪).  Getting triplets to work out might require a bit of zalgo, or we could just use subscripts (not ideal): ♪₃

**Comment by lostanlen** [comment]
I love this. I think it would be nice to accomodate a pure-text version though. wWth words: `quarter note`, `eigth note`, etc.

so as to be ASCII-friendly (and myopia-friendly 🧐 )

**Comment by bmcfee** [comment]
> I love this. I think it would be nice to accomodate a pure-text version though.

I think we could extend from the `unicode=False` support to do this.  The only question is what the right level of verbosity is?  I think having long-form names (`quarter note`) would be a bit too cluttered.  Is there a common short form for duration information?  Or would numbers make sense  (1/4, 1/2, 1/8, etc)?  That seems to be how humdrum and abc handle this.

**Comment by lostanlen** [comment]
yes, fraction is good too. much better than English text actually

**Comment by bmcfee** [comment]
A prototype using unicode... it's beautiful :smiling_face_with_tear: 

![image](https://user-images.githubusercontent.com/1190540/146962984-a7a94559-8ea9-4fd3-be30-97d5cb9b379b.png)

Oh well, it was a nice idea anyway.

**Comment by bmcfee** [comment]
Actually it looks like https://github.com/matplotlib/matplotlib/pull/20740 , if it gets merged, would make this possible.  It would bump our minimum supported matplotlib to at least 3.6 (when it comes out, etc), but I can live with that.

**Comment by bmcfee** [comment]
This issue is no longer blocked, but it will still depend on the 3.6 release of matplotlib.

**Comment by bmcfee** [comment]
Removing this from the 0.10 milestone - I don't think it's going to get done in the near future, and I'd rather not bump the minimum matplotlib version just for this anyway.

**Comment by bmcfee** [comment]
Bumping this to 1.0 milestone.

Per the comment in #1935 , SPEC0 would have us increment the minimum matplotlib to 3.8 (assuming a 2025.Q3 release), so we can rely on font fallback.

**Comment by bmcfee** [comment]
Hacked on this a bit this morning, and I think we probably have to scrap this idea for the time being, for two reasons:

1. Font fallback rather brittle, and we'd probably need to recommend users to install specific fonts (eg Noto Music) and tweak their matplotlib configs for this to work at all.
2. I haven't been able to think of a clean API for exposing control of the selected durations.  We can make this work easily enough for the defaults encoded in tempogram ratio, but if the user changes the duration ratios at all, we'd have to work out a way to propagate that to the display functions.  This would likely involve reverse engineering fractional floating point numbers, which is always a pain.  (It would be fine for dyadic ratios, but once we get into triplets or dotted notes, it will get messy with repeating mantissas...)

I think probably the best way forward on this is to just provide an advanced example notebook showing how to do it by hand.  I suspect the number of potential users for this functionality is in the single digits anyway, so that ought to be good enough.

**Comment by bmcfee** [comment]
Nudging this one: I wonder if the 3.11 matplotlib release helps us out on the fallback issues?  I have no idea now, but it seems plausible.
